src/token/src/canister.rs

```diff
@@ -207,38 +207,59 @@ impl TokenCanister {
     /// is not the global range of all transactions. The range [start, start + limit) here pertains to
     /// the transactions of user who. Implementations are allowed to return less TxRecords than
     /// requested to fend off DoS attacks.
+    ///
+    /// # Arguments
+    /// * `who` - The user to get transactions for.
+    /// * `start` - The index of the first transaction to return.
+    /// * `limit` - The number of transactions to return.
     #[query]
     fn getUserTransactions(&self, who: Principal, start: Nat, limit: Nat) -> Vec<TxRecord> {
-        let mut transactions = vec![];
-
         let limit_usize = limit.0.to_usize().unwrap_or(usize::MAX);
         if limit_usize > MAX_TRANSACTION_QUERY_LEN {
             ic_kit::ic::trap(&format!(
-                "Limit must be less then {}",
+                "Limit must be less than {}",
                 MAX_TRANSACTION_QUERY_LEN
             ));
         }
 
-        for tx in self.state.borrow().ledger.get_range(&start, &limit) {
-            if tx.from == who || tx.to == who || tx.caller == Some(who) {
-                transactions.push(tx.clone());
-            }
+        let length = self
+            .state
+            .borrow()
+            .ledger
+            .iter()
+            .filter(|tx| tx.from == who || tx.to == who || tx.caller == Some(who))
+            .count();
+
+        if start > length {
```

Comment:
again, no need to trap, just return empty.

Reply:
so if start is greater than length , we return empty vec ?

Reply:
usually with array indexing. arr[10..100) if the array index is greater than the length of the array, it will return up to the last element in the array. It's only indexing an element arr[100]  that will throw an error. So this shouldnt require any changes in the code.